build_dependency_tree.py

```diff
@@ -3,76 +3,202 @@
+        # We also need to ignore include statements made in comments
         include_statements = []
-        b_is_block_comment = False # Lines that start with /*
+        b_is_block_comment = False  # Lines that start with /*
         for line in f.readlines():
+            print(f"{line.strip()}")
             if b_is_block_comment:
-                if line.split()[0] == '*/':
+                if '*/' in line:
                     b_is_block_comment = False
                 else:
                     continue
             else:
-                if line.contains('/*'):
+                if '/*' in line:
                     b_is_block_comment = True
                     continue
-                elif line.split() == '//':
+                elif line.replace(' ', '')[0:2] == '//':
+                    continue
+                elif not line:  # empty line
                     continue
-                elif line.split()[0] == '#include':
-                    # match either the pattern `<...>` or `"..."`
-                    # re.findall will return a list of tuples `[(pattern 1, pattern 2), ...]`
-                    re_match = re.findall(r"<(.+?.)>|\"(.+?.)\"", line.split()[1])[0]
+                elif line.replace(' ', '')[0:8] == '#include' or \
+                        line.replace(' ', '')[0:9] == '%:include':
+                    # we always expect the included file to be given second
+                    # (required by standards)
+                    re_match = include_statement_pattern.match(line.split()[1])
                     # We only expect one entry in the list
-                    if not re_match[0]: # first matching group is empty
+                    if not re_match:  # first matching group is empty
+                        print(f'No header files found in line: {line}')
                         continue
+                    elif not re_match.group(1):
+                        # Check if match with `<...>` is None
+                        print(f"""Header file found
+                                {str(file.absolute())}:{re_match.group(2)}""")
+                        include_statements.append(
+                                extract_header_file_name(
+                                    re_match.group(2)))
                     else:
-                        include_statements.append(re_match[0])
-                else:
-                    continue
+                        # since we have a match and first group is non-empty we
+                        # apend
+                        print(f"""Header file found
+                                {str(file.absolute())}:{re_match.group(1)}""")
+                        include_statements.append(
+                                extract_header_file_name(
+                                    re_match.group(1)))
         return include_statements
+
+
+def output_dependency_tree_to_dot_file(
+        dep_tree: Dict[List[Union[Path, str]], str],
```

Comment:
```suggestion
        dep_tree: Dict[str, List[Union[Path, str]]],
```
keys cannot be a iterables

Reply:
keys are also innately, set like objects

Reply:
Then we would want the key to be a `Union[Path, str]`